testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/sso/remote/AbstractRemoteSingleSignOnTestCase.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2020, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.test.clustering.cluster.sso.remote;
+
+import org.infinispan.commons.test.skip.OS;
+import org.infinispan.commons.test.skip.SkipJunit;
+import org.infinispan.server.test.core.ServerRunMode;
+import org.infinispan.server.test.core.TestSystemPropertyNames;
+import org.infinispan.server.test.junit4.InfinispanServerRuleBuilder;
+import org.jboss.as.test.clustering.cluster.sso.AbstractSingleSignOnTestCase;
+import org.jgroups.util.Util;
+import org.junit.ClassRule;
+import org.junit.rules.TestRule;
+
+/**
+ * @author Radoslav Husar
+ */
+public abstract class AbstractRemoteSingleSignOnTestCase extends AbstractSingleSignOnTestCase {
+
+    // Disable on Windows until https://issues.redhat.com/browse/ISPN-12041 is fixed
+    @ClassRule
+    public static final TestRule INFINISPAN_SERVER_RULE = Util.checkForWindows() ? new SkipJunit(OS.WINDOWS) : InfinispanServerRuleBuilder
```

Comment:
Can we move INFINISPAN_SERVER_RULE to an interface?  That way we don't need to duplicate for every remote-cache-based test class hierarchy.

Reply:
It appears it's not possible, it needs to be a static field on the test class. Nevertheless, I extracted the `org.junit.rules.TestRule` into a separate method, so the setup is just 1 annotated field.